### DISCUSSION TITLE
Migrating from BitBucket to GitHub Cloud (JIRA)

### DISCUSSION THREAD

**Discussion by APAWLTRW** [community/community#45933] [New to GitHub]
Any resources out there that can assist with guidance on this. I'm not a developer but an App Admin and I have been tasked with this migration. I am looking for best practices - any tips that someone such as myself may benefit from in order to make this as painless as possible.

Assistance is greatly appreciated.

**Comment by wcontreras** [comment]
Hi brother
I've had a fight lately with something like this.

What worked perfect for me was cloning the entire bitbucket repository on my machine and then uploading it to github since I couldn't find an automated mechanism.

With these steps I understand that they can be useful if you still have not found anything.

1. Clone an exact copy of your remote repository to your machine with the following command. 
`git clone --mirror REMOTE-URL-BITBUCKET`
2. Create your repository on github. I recommend you use the same name so as not to be confused later.
3. Change the URL of the remote bitbucket repository to the one you recently created on github with the command 
`git remote set-url origin REMOTE-URL-GITHUB`
4. After doing the above, simply upload the project with the following command
`git push --mirror`

With these steps you upload all the changes you have had over time without losing any branch or commit

If someone suggests something more effective I will be more than grateful.

**Reply by APAWLTRW** [reply]
So we have channel 0 errors - could not figure it out. Chose to download from BitBucket and then upload to Git but we are getting the following error - Committing the files to the repository failed.

Not sure if there are any logs other then the Security log - which does not tell me what is happening.

Anyone else experience this?

**Comment by APAWLTRW** [comment]
So this works if I commit the contents of the Repo Folder but not from the Folder level (in Windows).

Odd?

**Comment by APAWLTRW** [comment]
If I download a repo does in include all the commits and branches?

**Comment by Ant0wan** [comment]
It looks not like proper migration but just mirroring. How do you handle opened PRs ? Webhooks ? etc
